### ISSUE TITLE
How to add ImageMagic when deploying using mupx?

### ISSUE THREAD

**Issue by scsirdx** [arunoda/meteor-up-legacy#635]
Recently i switched mup to mupx, and now my app can't use ImageMagic. I think, its because of docker image. How can i install imagemagic into container with app?

**Comment by MasterJames** [comment]
Are you talking about this stuff...
http://www.imagemagick.org/script/binary-releases.php
Are you using 
npm install imagemagick
to install it?
What is your build platform, and target server platform/image?

**Comment by scsirdx** [comment]
Maybe i don't understand clearly new deployment process.
With mup i had installed imagemagic on server and app can use it after deploy.
mupx deploy app in docker container, so it can't use imagemagic installed on server.

Target platforms: server, browser.

**Comment by MasterJames** [comment]
Oh I see. Is there a way to use "npm install imagemagick", possibly?
I wondered before what your Ubuntu version for the server and if your deploying from windows or what? 
I'm sorry I don't have any other ideas on this myself (hopefully some else here has an idea) other then going back to mup or changing things to work maybe differently through the npm route or something.

**Comment by scsirdx** [comment]
I have little experience in docker. In fact, this is a first time i use it in production.
I think, that anyone who plan to use image upload and autoresize (cfs:graphicsmagick for example) will face this problem. So it will be good if one of mupx developers will tell how to use cfs:graphicsmagick with mupx.

**Comment by miri-am** [comment]
same problem here

**Comment by miri-am** [comment]
this is related to  #308

**Comment by MasterJames** [comment]
I hope someone else can suggest a script tag to install it but otherwise I think you're looking at a custom docker image then what is available at the provider. 
https://coreos.com/os/docs/latest/customizing-docker.html
I'll keep looking around in the meantime.

**Comment by miri-am** [comment]
Yes, it would be nice to have documentation on how to add npm modules. Sorry, so far I have no experiences with docker but would be interesting to know:
- Should we specify our own docker image? How can we do that and still be able to update mupx?
- Should there be an option within mupx (like in mup.json)? 
- Should mupx automatically check which modules need to be included?

**Comment by MasterJames** [comment]
I think you can manually add stuff to a docker image with 'pull' and then 'push' those changes.
https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-getting-started
so pull this...
https://github.com/jujhars13/docker-imagemagick
...and push it back.

related ...
https://github.com/arunoda/meteor-up/issues/546

**Comment by MasterJames** [comment]
Maybe one could simply insert the...
docker pull jujhars13/docker-imagemagick
into the linux start.sh before line 19 and do it dependent on an optional variable.
https://github.com/arunoda/meteor-up/blob/mupx/templates/linux/start.sh

**Comment by miri-am** [comment]
@MasterJames thanks. Finally found and used this stackoverflow answer, which modifies the start.sh script and installs graphicsmagick: http://stackoverflow.com/a/31903981/3223028. Works for me now.

**Comment by sakulstra** [comment]
same problem here,
:+1: some integrated solution without hacking the start.sh would be nice ;)

**Comment by advany** [comment]
I tried the start.sh solution but I get this error:

Started TaskList: Deploy app 'islamagenda' (linux)
[188.166.109.203] - Uploading bundle
[188.166.109.203] - Uploading bundle: SUCCESS
[188.166.109.203] - Sending environment variables
[188.166.109.203] - Sending environment variables: SUCCESS
[188.166.109.203] - Initializing start script
[188.166.109.203] - Initializing start script: SUCCESS
[188.166.109.203] - Invoking deployment process
[188.166.109.203] - Invoking deployment process: SUCCESS
[188.166.109.203] - Verifying deployment
[188.166.109.203] x Verifying deployment: FAILED

```
-----------------------------------STDERR-----------------------------------
 to the latest stable version, run:
npm WARN deprecated 
npm WARN deprecated   npm -g install npm@latest
npm WARN deprecated 
npm WARN deprecated (Depending on how Node.js was installed on your system, you
npm WARN deprecated may need to prefix the preceding commands with `sudo`, or if
npm WARN deprecated on Windows, run them from an Administrator prompt.)
npm WARN deprecated 
npm WARN deprecated If you're running the version of npm bundled with
npm WARN deprecated Node.js 0.10 LTS, be aware that the next version of 0.10 LTS
npm WARN deprecated will be bundled with a version of npm@2, which has some small
npm WARN deprecated backwards-incompatible changes made to `npm run-script` and
npm WARN deprecated semver behavior.
npm WARN package.json meteor-dev-bundle@0.0.0 No description
npm WARN package.json meteor-dev-bundle@0.0.0 No repository field.
npm WARN package.json meteor-dev-bundle@0.0.0 No README data
=> Starting meteor app on port:80

=> Redeploying previous version of the app

-----------------------------------STDOUT-----------------------------------

To see more logs type 'mup logs --tail=50'

----------------------------------------------------------------------------
```

**Comment by MasterJames** [comment]
What version of nodejs is setup in the settings? 
It should be 0.10.40 for Meteor 1.2

**Comment by PhilippSpo** [comment]
+1

**Comment by d-i-b** [comment]
Same issue @advany any progress?

**Comment by advany** [comment]
I found mine error
had fsdebug as true

**Comment by advany** [comment]
found also an other issue. cfs:filesystem doesn;t work. Remove it and add cfs:gridfs

**Comment by xvendo** [comment]
Please use MUP (the development version) there you can add images. I had the same issue and was using MUPX. It was not possible to fix the graphicsmagick integration problem. I installed latest MUP and added an docker image. It worked like a charm. You can lookup many images over here https://hub.docker.com/. Hope that helped someone. Feel free to get in touch if you need more info.
